internal/flink/command_artifact_create.go

```diff
@@ -0,0 +1,102 @@
+package flink
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	connectcustompluginv1 "github.com/confluentinc/ccloud-sdk-go-v2/connect-custom-plugin/v1"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+	"github.com/confluentinc/cli/v3/pkg/output"
+)
+
+type pluginCreateOut struct {
+	Name           string `human:"Name" serialized:"name"`
+	Id             string `human:"Plugin ID" serialized:"id"`
+	ConnectorClass string `human:"Version ID" serialized:"connector_class"`
+	ContentFormat  string `human:"Content Format" serialized:"content_format"`
+	Scope          string `human:"Scope"`
+	ErrorTrace     string `human:"Error Trace,omitempty" serialized:"error_trace,omitempty"`
+}
+
+func (c *command) newCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create <name>",
+		Short: "Create a Flink UDF artifact.",
+		Args:  cobra.ExactArgs(1),
+		RunE:  c.createArtifact,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Create Flink UDF artifact "my-flink-artifact".`,
+				Code: "confluent flink artifact create my-flink-artifact --artifact-file /Users/xyz/Documents/config/plugin.jar",
+			},
+		),
+	}
+
+	cmd.Flags().String("artifact-file", "", "ZIP/JAR Flink UDF artifact file.")
+	cmd.Flags().String("description", "", "Description of Flink UDF artifact.")
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	cobra.CheckErr(cmd.MarkFlagRequired("artifact-file"))
+	cobra.CheckErr(cmd.MarkFlagFilename("artifact-file", "jar"))
```

Comment:
The flag description implies this can also be a `.zip` file

Reply:
good catch. updated.

Reply:
Did you remember to commit this change?

Reply:
@brianstrauch flink artifact will be JAR only for EA

Reply:
updated by removing `zip`